exercise/static/exercise/group.js

```diff
@@ -46,13 +46,13 @@
         });
 
         this.ui = $('<div class="submit-group-selector input-group col-md-6">'
-          + '<select name="_aplus_group" class="form-control"></select>'
+          + '<select name="__aplus__" class="form-control"></select>'
           + '<span class="input-group-btn"></span>'
           + '</div>');
         var list = this.ui.find('select');
         for (var i = 0; i < this.groups.length; i++) {
           var group = this.groups[i];
-          list.append($('<option value="' + group.id + '" data-group-size="' + group.size + '">' + group.text + '</option>'));
+          list.append($('<option value="{&quot;group&quot;:' + group.id + '}" data-group-size="' + group.size + '">' + group.text + '</option>'));
```

Comment:
I think this is probably a bit too simplified change. I think you need to build the `__aplus__` in the javascript so we can bundle other things like the language in there too.

Reply:
Yeah that would make sense, but there isn't any language yet for the submission,  I don't know if there is a point in setting the language in the js, when it's going to be done when the submission is created. It would look nicer if  `__aplus__` is built but wouldn't it still be only the group in there? (at least for now)

Reply:
The language should be kept as part of the data somehow, as the user language might change between GET and POST, If the language changes between GET and POST, then the submission is graded with different language than what was shown to the user. This is the problem the `__grader_lang` tries to solve for programming exercises.